documentation/jetty/modules/code/examples/src/main/java/org/eclipse/jetty/docs/programming/ArchitectureDocs.java

```diff
@@ -75,4 +79,85 @@ public void virtualVirtualThreads()
         HttpClient httpClient = new HttpClient(new HttpClientTransportOverHTTP(clientConnector));
         // end::virtualVirtual[]
     }
+
+    public static class EitherTask implements Invocable.Task
+    {
+        private final TryExecutor executor;
+        private final Deque<Invocable.Task> subTasks = new ConcurrentLinkedDeque<>();
+
+        public EitherTask(TryExecutor executor)
+        {
+            this.executor = executor;
+        }
+
+        @Override
+        public InvocationType getInvocationType()
+        {
+            return InvocationType.EITHER;
+        }
+
+        public void offer(Task task)
+        {
+            subTasks.add(task);
+        }
+
+        @Override
+        public void run()
+        {
+            if (Invocable.isNonBlockingInvocation())
+            {
+                for (Task subTask = subTasks.pollFirst(); subTask != null; subTask = subTasks.pollFirst())
+                {
+                    switch (Invocable.getInvocationType(subTask))
+                    {
+                        case NON_BLOCKING, EITHER ->
+                            // Do not defer a NON-BLOCKING or EITHER task
+                            // We do not need to use invokeNonBlocking, as we are already invoked via that method
+                            subTask.run();
+
+                        case BLOCKING ->
+                            // Defer the BLOCKING task as this thread may not block
+                            executor.execute(subTask);
+                    }
+                }
+            }
+            else
+            {
+                for (Task subTask = subTasks.pollFirst(); subTask != null; subTask = subTasks.pollFirst())
+                {
+                    switch (Invocable.getInvocationType(subTask))
+                    {
+                        case NON_BLOCKING ->
+                            // Do not defer a NON-BLOCKING task
+                            subTask.run();
+
+                        case EITHER ->
+                        {
+                            // Do not defer an EITHER task
+                            if (executor.tryExecute(this))
+                                // A reserved thread is consuming any remaining subtasks, so we can call the subtask directly,
+                                // and it may block without causing head-of-line blocking.
+                                subTask.run();
+                            else
+                                // There is no reserved thread to consume remaining subtasks, so we invoke this
+                                // subtask as non-blocking so that we avoid head-of-line blocking.
```

Comment:
```suggestion
                                // There is no reserved thread to consume remaining subtasks; invoke this
                                // subtask as non-blocking to avoid head-of-line blocking.
```